File: src/test/java/dev/retrotv/crypt/owe/argon2/Argon2Test.java
package dev.retrotv.crypt.owe.argon2;

import dev.retrotv.crypt.owe.OWETest;
import dev.retrotv.crypt.owe.Password;
import dev.retrotv.crypt.owe.PasswordWithSalt;
import dev.retrotv.crypt.owe.crc.CRC32;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

public class Argon2Test extends OWETest {

    @Test
    @DisplayName("Argon2 password encode 테스트")
    void passwordEncrypt() {
        passwordEncryptAndMatchesTest(new Argon2());
    }
}

File: cryptography-password/src/test/kotlin/dev/retrotv/crypto/password/argon2/Argon2Test.kt
package dev.retrotv.crypto.password.argon2

import dev.retrotv.crypto.password.KDFTest
import org.junit.jupiter.api.DisplayName
import kotlin.test.Test
import kotlin.test.assertNotEquals
import kotlin.test.assertTrue

class Argon2Test {
    private val test = KDFTest()
    private val password = "The lazy dog jumps over the quick brown fox"

    @Test
    @DisplayName("Argon2 암호화 테스트")
    fun test_argon2() {
        var argon2 = Argon2()
        test.test_kdf(argon2)

        /* --------------------------------------------------------------- */

        argon2 = Argon2(16, 16, 2, 1 shl 14, 2)
        test.test_kdf(argon2)
    }
}